instance.tf

```diff
@@ -3,45 +3,85 @@ resource "google_sql_database_instance" "instance" {
   name                 = var.instance_name
   deletion_protection  = var.deletion_protection
   master_instance_name = var.primary_instance_name
+
   settings {
+    activation_policy     = "ALWAYS"
     availability_type     = var.highly_available == true ? "REGIONAL" : "ZONAL"
     disk_autoresize       = var.storage_autoresize
     disk_autoresize_limit = var.storage_limit
     disk_size             = local.storage_size
     disk_type             = "PD_SSD"
+    pricing_plan          = "PER_USE"
     tier                  = local.tier
     user_labels           = local.labels
+
     backup_configuration {
-      enabled  = local.backup_config.enabled
-      location = local.backup_config.location
+      enabled                        = local.backup_config.enabled
+      start_time                     = local.backup_config.start_time
+      point_in_time_recovery_enabled = local.backup_config.point_in_time_recovery_enabled
+      location                       = local.backup_config.location
+      transaction_log_retention_days = local.backup_config.transaction_log_retention_days
+      dynamic "backup_retention_settings" {
+        for_each = compact([local.backup_config.retained_backups])
+
+        content {
+          retained_backups = local.backup_config.retained_backups
+          retention_unit   = "COUNT"
+        }
+      }
     }
+
     dynamic "database_flags" {
       for_each = var.flags
       iterator = flag
+
       content {
         name  = flag.key
         value = flag.value
       }
     }
+
+    dynamic "insights_config" {
+      for_each = var.insights_config == null ? [] : [0]
+
+      content {
+        query_insights_enabled  = var.insights_config.query_insights_enabled
+        query_string_length     = lookup(var.insights_config, "query_string_length", 1024)
+        record_application_tags = lookup(var.insights_config, "record_application_tags", false)
+        record_client_address   = lookup(var.insights_config, "record_client_address", false)
+      }
+    }
+
     ip_configuration {
       ipv4_enabled = true
-      require_ssl  = true
+      #private_network = null
```

Comment:
Implement what exactly? As for what to (not) implement: just look at what we're currently customizing. If more custom stuff is needed later on, it'll need to get added. No need to run ahead of ourselves.

Reply:
Private network support. At least one of `private_network` and `ipv4_enabled` has to be enabled. Currently all of our instances have ipv4 enabled. Might be interesting to use private network by the time we set up a new k8s cluster with a [Shared VPC network](https://cloud.google.com/vpc/docs/shared-vpc#shared_vpc_networks) (this came up when migrating MySQL from AWS to GCP). Other than that I don't see any reason for us to use this.

Reply:
Explicitly setting it to `null` didn't work I guess? It's rather annoying that some stuff can't be explicitly set to the same value as it'd default to when not specified. Not specifying it would allow drift as enabling private network manually, wouldn't get detected by terraform :disappointed:

Reply:
Haven't tried that actually. I'll give it a go.

Reply:
Apparently it's actually been set for at least on database (casco prod), which means I'll create a variable for this.
Should I automatically set `ipv4_enabled` to true when `private_network` isn't specified? Or should I leave it to `terraform plan` to error on it?

Reply:
I'm leaning towards leaving that to terraform itself.